tools/SetupFlow/DevHome.SetupFlow/Views/AddRepoDialog.xaml.cs

```diff
@@ -337,21 +340,99 @@ private async void AddRepoContentDialog_PrimaryButtonClick(ContentDialog sender,
                 SwitchToRepoPage(repositoryProviderName);
             }
         }
+        else if (AddRepoViewModel.CurrentPage == PageKind.SearchFields)
+        {
+            args.Cancel = true;
+            Dictionary<string, string> searchInput = new();
+            foreach (var searchBox in ShowingSearchTermsGrid.Children)
+            {
+                if (searchBox is AutoSuggestBox suggestBox)
+                {
+                    searchInput.Add(suggestBox.Header as string, suggestBox.Text);
+                }
+            }
+
+            // switching to the repo page causes repos to be queried.
+            SwitchToRepoPage(AddRepoViewModel.ProviderNames[0], searchInput);
+        }
+    }
+
+    private void SwitchToSelectSearchTermsPage()
+    {
+        AddRepoViewModel.ChangeToSelectSearchTermsPage();
+        AddRepoViewModel.FolderPickerViewModel.ShouldShowFolderPicker = Visibility.Collapsed;
+        EditDevDriveViewModel.ShowDevDriveInformation = Visibility.Collapsed;
+        IsPrimaryButtonEnabled = true;
+    }
+
+    private async void SwitchToRepoPage(string repositoryProviderName, Dictionary<string, string> inputValues)
+    {
+        await AddRepoViewModel.GetAccountsAsync(repositoryProviderName, LoginUIContent);
+        if (AddRepoViewModel.Accounts.Any())
+        {
+            AddRepoViewModel.ChangeToRepoPage(inputValues);
+            AddRepoViewModel.FolderPickerViewModel.ShowFolderPicker();
+            EditDevDriveViewModel.ShowDevDriveUIIfEnabled();
+            AddRepoViewModel.SelectedAccount = AddRepoViewModel.Accounts.First();
+            AddRepoViewModel.ShouldEnablePrimaryButton = false;
+        }
     }
 
     private async void SwitchToRepoPage(string repositoryProviderName)
     {
         await AddRepoViewModel.GetAccountsAsync(repositoryProviderName, LoginUIContent);
         if (AddRepoViewModel.Accounts.Any())
         {
-            AddRepoViewModel.ChangeToRepoPage();
+            AddRepoViewModel.ChangeToRepoPage(new());
             AddRepoViewModel.FolderPickerViewModel.ShowFolderPicker();
             EditDevDriveViewModel.ShowDevDriveUIIfEnabled();
             AddRepoViewModel.SelectedAccount = AddRepoViewModel.Accounts.First();
             AddRepoViewModel.ShouldEnablePrimaryButton = false;
         }
     }
 
+    private void FilterSuggestions(AutoSuggestBox sender, AutoSuggestBoxTextChangedEventArgs args)
+    {
+        sender.ItemsSource = _searchFieldsAndValues[sender.Header.ToString()].Where(x => x.Contains(sender.Text));
+        return;
+    }
+
+    private async void HyperlinkButton_Click(object sender, RoutedEventArgs e)

```

Comment:
Can this method get a more descriptive name?

Reply:
Changed.